### PR TITLE
Adds ProvideItem and ProvideItems RequestStateController Types

### PR DESCRIPTION
**Pull request by Twigz**
## What it Does
Adds provideItem and provideItems request state controllers.

### CODE REVIEW

File: Provider.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -1,34 +1,32 @@
...
+      "identity" : "persister",
+      "kind" : "remoteSourceControl",
+      "location" : "https://github.com/Lickability/Persister",
+      "state" : {
+        "branch" : "main",
+        "revision" : "ac8f39433b3ced5df28f6c414d06ddc50a57ba7c"
+      }
+    }
+  ],
+  "version" : 2
```
Comment (mliberatore) [inline Provider.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:31]: Interesting, looks like the format changed (version: 2)?

File: Sources/Provider/ProvideItemRequestStateController.swift [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                    switch error {
+                    case let .networkError(networkError):
+                        return networkError
+                    case let .decodingError(error):
+                        if let decodingError = error as? DecodingError {
+                            print(decodingError)
+                        }
+                        
+                        return error as? LocalizedError
+                    default:
```
Comment (mliberatore) [inline Sources/Provider/ProvideItemRequestStateController.swift:58]: Any chance I can get you to be exhaustive here (still `return error`) so newly added cases produce a compiler error?

File: Sources/Provider/ProvideItemsRequestStateController.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                return nil
+            case let .completed(result):
+                switch result {
+                case .success:
+                    return nil
+                case let .failure(error):
+                    switch error {
+                    case let .networkError(networkError):
+                        return networkError
+                    default:
```
Comment (mliberatore) [inline Sources/Provider/ProvideItemsRequestStateController.swift:52]: Same here

**Review by mliberatore — APPROVED** [review]
Thanks for updating those property names here, too

File: Provider.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (Twigz) [reply Provider.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:31]: Sure did!

**Pull request merged**